### ISSUE TITLE
co-expression p/q-values

### ISSUE THREAD

**Issue by jjgao** [cBioPortal/cbioportal#4248]
- [x] in web API, remove Pearson's correlation scores 
- [x] in web API, calculate p/q values of [Spearman's correlation](http://commons.apache.org/proper/commons-math/javadocs/api-3.3/org/apache/commons/math3/stat/correlation/PearsonsCorrelation.html)
- [x] co-expression tab: add p value and q value columns and order by p value
- [x] co-expression tab: remove Pearson's correlation scores in the table.

**Comment by jjgao** [comment]
@ersinciftci I am wondering whether the performance would be good with [the Apache Spearman's correlation p-value calculation](http://commons.apache.org/proper/commons-math/javadocs/api-3.3/org/apache/commons/math3/stat/correlation/PearsonsCorrelation.html)

**Comment by ersinciftci** [comment]
@jjgao we already use Apache Commons Math for Pearson and Spearman calculations

**Comment by adamabeshouse** [comment]
what is a vocation plot?

**Comment by jjgao** [comment]
@adamabeshouse LOL -- I mean vocano plot. I guess I need vacations.

**Comment by jjgao** [comment]
@ersinciftci great. Hopefully it's trivial to add the p-value. Could you test the performance?

**Comment by jjgao** [comment]
@adamabeshouse @ersinciftci this one doesn't have to be in 1.14.0.

**Comment by jjgao** [comment]
It looks like a user is asking for it now:

https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/cbioportal/dPw6HS0Ko5A/Y1t7TfbwBQAJ

@ersinciftci if you have the bandwidth, could you look into this and see if it's straightforward to extend the API? Once that's done, @adamabeshouse and/or you could change the frontend. Thanks.

**Comment by jjgao** [comment]
@ersinciftci could you make the change to the API? Could you also do the frontend change too? Or should @adamabeshouse work on it? I was thinking maybe the same layout should be used here as Enrichments (Volcano plots and scatter plots on the left side and table on the right side)

**Comment by ersinciftci** [comment]
I don't know much about this tab's frontend, I think would be better if @adamabeshouse does the frontend change.

**Comment by adamabeshouse** [comment]
I can make the frontend change once the API is done

**Comment by jjgao** [comment]
Sounds good! Let's prioritize this. There is another request:

https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/cbioportal/T6Z3TJg_aiU

**Comment by adamabeshouse** [comment]
did you close by mistake @jjgao ?

**Comment by jjgao** [comment]
@adamabeshouse sorry by mistake.

**Comment by jjgao** [comment]
I think the frontend part is not done, right?

**Comment by ersinciftci** [comment]
Ah, yeah, will just unassign myself for now.

**Comment by adamabeshouse** [comment]
@jjgao where do you want to put the volcano plot? and how big?
![image](https://user-images.githubusercontent.com/636232/44240671-4f6ffc80-a18d-11e8-9031-4f482f976eaa.png)

**Comment by jjgao** [comment]
@adamabeshouse I think we can use the same layout as `enrichments` tab. We can move the current scatter plot to left too (replacing the box plot in the screenshot below; maybe moving the legend under the plot).

![image](https://user-images.githubusercontent.com/840895/45440116-6e2cba80-b689-11e8-93c2-bf047ab925ba.png)

**Comment by inodb** [comment]
@jjgao i moved the volcano plot item to a separate issue, so we can schedule that for a diff release

**Comment by inodb** [comment]
this is all in 1.17 branch now
